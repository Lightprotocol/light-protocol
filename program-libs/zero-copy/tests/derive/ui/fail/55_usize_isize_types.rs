// Edge case: usize and isize types
#![cfg(feature = "mut")]
use light_zero_copy_derive::{ZeroCopy, ZeroCopyMut};

#[derive(ZeroCopy, ZeroCopyMut)]
#[repr(C)]
pub struct SizeTypes {
    pub unsigned_size: usize,
    pub signed_size: isize,
    pub vec_usize: Vec<usize>,
    pub opt_isize: Option<isize>,
    pub array_usize: [usize; 8],
}

fn main() {}
