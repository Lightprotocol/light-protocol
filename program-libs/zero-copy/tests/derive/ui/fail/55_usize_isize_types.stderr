error[E0277]: the trait bound `usize: ZeroCopyAt<'a>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:10
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |          ^^^^^^^^ the trait `ZeroCopyAt<'a>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `ZeroCopy` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyAt<'a>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:10
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |          ^^^^^^^^ the trait `ZeroCopyAt<'a>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `ZeroCopy` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `usize: ZeroCopyNew<'static>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyNew<'static>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `ZeroCopyMut` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyNew<'static>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyNew<'static>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `ZeroCopyMut` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyNew<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyNew<'_>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8
  = note: required for `Option<isize>` to implement `ZeroCopyNew<'static>`
note: required because it appears within the type `SizeTypesConfig`
 --> tests/ui/fail/55_usize_isize_types.rs:7:12
  |
7 | pub struct SizeTypes {
  |            ^^^^^^^^^
note: required by a bound in `Clone`
 --> $RUST/core/src/clone.rs
  |
  | pub trait Clone: Sized {
  |                  ^^^^^ required by this bound in `Clone`
  = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `usize: ZeroCopyAtMut<'a>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyAtMut<'a>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyAtMut<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `ZeroCopyMut` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyAtMut<'a>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyAtMut<'a>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyAtMut<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `ZeroCopyMut` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyNew<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:7:12
  |
7 | pub struct SizeTypes {
  |            ^^^^^^^^^ the trait `ZeroCopyNew<'_>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8
  = note: required for `Option<isize>` to implement `ZeroCopyNew<'static>`
note: required because it appears within the type `SizeTypesConfig`
 --> tests/ui/fail/55_usize_isize_types.rs:7:12
  |
7 | pub struct SizeTypes {
  |            ^^^^^^^^^
note: required by a bound in `light_zero_copy::ZeroCopyNew::ZeroCopyConfig`
 --> $WORKSPACE/program-libs/zero-copy/src/traits/zero_copy_new.rs
  |
  |     type ZeroCopyConfig;
  |     ^^^^^^^^^^^^^^^^^^^^ required by this bound in `ZeroCopyNew::ZeroCopyConfig`

error[E0277]: the trait bound `usize: ZeroCopyAt<'a>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:10
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |          ^^^^^^^^ the trait `ZeroCopyAt<'a>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyAt<'a>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:10
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |          ^^^^^^^^ the trait `ZeroCopyAt<'a>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `usize: ZeroCopyAt<'a>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:10
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |          ^^^^^^^^ the trait `ZeroCopyAt<'a>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `PartialEq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyAt<'a>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:10
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |          ^^^^^^^^ the trait `ZeroCopyAt<'a>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `PartialEq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `usize: ZeroCopyAt<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:10
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |          ^^^^^^^^ the trait `ZeroCopyAt<'_>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyAt<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:10
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |          ^^^^^^^^ the trait `ZeroCopyAt<'_>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `usize: ZeroCopyAt<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:8:24
  |
8 |     pub unsigned_size: usize,
  |                        ^^^^^ the trait `ZeroCopyAt<'_>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8

error[E0277]: the trait bound `isize: ZeroCopyAt<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:9:22
  |
9 |     pub signed_size: isize,
  |                      ^^^^^ the trait `ZeroCopyAt<'_>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8

error[E0277]: the trait bound `usize: ZeroCopyAt<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:10
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |          ^^^^^^^^ the trait `ZeroCopyAt<'_>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: required for `Vec<usize>` to implement `ZeroCopyAt<'_>`
  = note: this error originates in the derive macro `ZeroCopy` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyAt<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:10
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |          ^^^^^^^^ the trait `ZeroCopyAt<'_>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: required for `Option<isize>` to implement `ZeroCopyAt<'_>`
  = note: this error originates in the derive macro `ZeroCopy` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `usize: ZeroCopyAt<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:8:9
  |
8 |     pub unsigned_size: usize,
  |         ^^^^^^^^^^^^^ the trait `ZeroCopyAt<'_>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8

error[E0277]: the trait bound `isize: ZeroCopyAt<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:9:9
  |
9 |     pub signed_size: isize,
  |         ^^^^^^^^^^^ the trait `ZeroCopyAt<'_>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyAt<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8

error[E0277]: the trait bound `usize: ZeroCopyAt<'_>` is not satisfied
  --> tests/ui/fail/55_usize_isize_types.rs:10:9
   |
10 |     pub vec_usize: Vec<usize>,
   |         ^^^^^^^^^ the trait `ZeroCopyAt<'_>` is not implemented for `usize`
   |
   = help: the following other types implement trait `ZeroCopyAt<'a>`:
             i16
             i32
             i64
             u16
             u32
             u64
             u8
   = note: required for `Vec<usize>` to implement `ZeroCopyAt<'_>`

error[E0277]: the trait bound `isize: ZeroCopyAt<'_>` is not satisfied
  --> tests/ui/fail/55_usize_isize_types.rs:11:9
   |
11 |     pub opt_isize: Option<isize>,
   |         ^^^^^^^^^ the trait `ZeroCopyAt<'_>` is not implemented for `isize`
   |
   = help: the following other types implement trait `ZeroCopyAt<'a>`:
             i16
             i32
             i64
             u16
             u32
             u64
             u8
   = note: required for `Option<isize>` to implement `ZeroCopyAt<'_>`

error[E0277]: the trait bound `usize: ZeroCopyNew<'static>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyNew<'static>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyNew<'static>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyNew<'static>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `usize: ZeroCopyNew<'static>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyNew<'static>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyNew<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyNew<'_>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8
  = note: required for `Option<isize>` to implement `ZeroCopyNew<'static>`
note: required because it appears within the type `SizeTypesConfig`
 --> tests/ui/fail/55_usize_isize_types.rs:7:12
  |
7 | pub struct SizeTypes {
  |            ^^^^^^^^^
  = note: the return type of a function must have a statically known size
  = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `usize: ZeroCopyNew<'static>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyNew<'static>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `PartialEq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyNew<'static>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyNew<'static>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `PartialEq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `usize: ZeroCopyAtMut<'a>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyAtMut<'a>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyAtMut<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyAtMut<'a>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyAtMut<'a>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyAtMut<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: this error originates in the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `usize: ZeroCopyAtMut<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:8:24
  |
8 |     pub unsigned_size: usize,
  |                        ^^^^^ the trait `ZeroCopyAtMut<'_>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyAtMut<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8

error[E0277]: the trait bound `isize: ZeroCopyAtMut<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:9:22
  |
9 |     pub signed_size: isize,
  |                      ^^^^^ the trait `ZeroCopyAtMut<'_>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyAtMut<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8

error[E0277]: the trait bound `usize: ZeroCopyAtMut<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyAtMut<'_>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyAtMut<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: required for `Vec<usize>` to implement `ZeroCopyAtMut<'_>`
  = note: this error originates in the derive macro `ZeroCopyMut` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `isize: ZeroCopyAtMut<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyAtMut<'_>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyAtMut<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8
  = note: required for `Option<isize>` to implement `ZeroCopyAtMut<'_>`
  = note: this error originates in the derive macro `ZeroCopyMut` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `usize: ZeroCopyAtMut<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:8:9
  |
8 |     pub unsigned_size: usize,
  |         ^^^^^^^^^^^^^ the trait `ZeroCopyAtMut<'_>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyAtMut<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8

error[E0277]: the trait bound `isize: ZeroCopyAtMut<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:9:9
  |
9 |     pub signed_size: isize,
  |         ^^^^^^^^^^^ the trait `ZeroCopyAtMut<'_>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyAtMut<'a>`:
            i16
            i32
            i64
            u16
            u32
            u64
            u8

error[E0277]: the trait bound `usize: ZeroCopyAtMut<'_>` is not satisfied
  --> tests/ui/fail/55_usize_isize_types.rs:10:9
   |
10 |     pub vec_usize: Vec<usize>,
   |         ^^^^^^^^^ the trait `ZeroCopyAtMut<'_>` is not implemented for `usize`
   |
   = help: the following other types implement trait `ZeroCopyAtMut<'a>`:
             i16
             i32
             i64
             u16
             u32
             u64
             u8
   = note: required for `Vec<usize>` to implement `ZeroCopyAtMut<'_>`

error[E0277]: the trait bound `isize: ZeroCopyAtMut<'_>` is not satisfied
  --> tests/ui/fail/55_usize_isize_types.rs:11:9
   |
11 |     pub opt_isize: Option<isize>,
   |         ^^^^^^^^^ the trait `ZeroCopyAtMut<'_>` is not implemented for `isize`
   |
   = help: the following other types implement trait `ZeroCopyAtMut<'a>`:
             i16
             i32
             i64
             u16
             u32
             u64
             u8
   = note: required for `Option<isize>` to implement `ZeroCopyAtMut<'_>`

error[E0277]: the trait bound `usize: ZeroCopyNew<'static>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:8:24
  |
8 |     pub unsigned_size: usize,
  |                        ^^^^^ the trait `ZeroCopyNew<'static>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8

error[E0277]: the trait bound `isize: ZeroCopyNew<'static>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:9:22
  |
9 |     pub signed_size: isize,
  |                      ^^^^^ the trait `ZeroCopyNew<'static>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8

error[E0277]: the trait bound `usize: ZeroCopyNew<'_>` is not satisfied
  --> tests/ui/fail/55_usize_isize_types.rs:10:20
   |
10 |     pub vec_usize: Vec<usize>,
   |                    ^^^^^^^^^^ the trait `ZeroCopyNew<'_>` is not implemented for `usize`
   |
   = help: the following other types implement trait `ZeroCopyNew<'a>`:
             u16
             u32
             u64
             u8
   = note: required for `Vec<usize>` to implement `ZeroCopyNew<'static>`

error[E0277]: the trait bound `isize: ZeroCopyNew<'_>` is not satisfied
  --> tests/ui/fail/55_usize_isize_types.rs:11:20
   |
11 |     pub opt_isize: Option<isize>,
   |                    ^^^^^^^^^^^^^ the trait `ZeroCopyNew<'_>` is not implemented for `isize`
   |
   = help: the following other types implement trait `ZeroCopyNew<'a>`:
             u16
             u32
             u64
             u8
   = note: required for `Option<isize>` to implement `ZeroCopyNew<'static>`

error[E0277]: the trait bound `isize: ZeroCopyNew<'_>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^ the trait `ZeroCopyNew<'_>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8
  = note: required for `Option<isize>` to implement `ZeroCopyNew<'static>`
note: required because it appears within the type `SizeTypesConfig`
 --> tests/ui/fail/55_usize_isize_types.rs:7:12
  |
7 | pub struct SizeTypes {
  |            ^^^^^^^^^
  = note: this error originates in the derive macro `ZeroCopyMut` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `usize: ZeroCopyNew<'a>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:8:24
  |
8 |     pub unsigned_size: usize,
  |                        ^^^^^ the trait `ZeroCopyNew<'a>` is not implemented for `usize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8

error[E0277]: the trait bound `isize: ZeroCopyNew<'a>` is not satisfied
 --> tests/ui/fail/55_usize_isize_types.rs:9:22
  |
9 |     pub signed_size: isize,
  |                      ^^^^^ the trait `ZeroCopyNew<'a>` is not implemented for `isize`
  |
  = help: the following other types implement trait `ZeroCopyNew<'a>`:
            u16
            u32
            u64
            u8

error[E0277]: the trait bound `usize: ZeroCopyNew<'_>` is not satisfied
  --> tests/ui/fail/55_usize_isize_types.rs:10:20
   |
10 |     pub vec_usize: Vec<usize>,
   |                    ^^^^^^^^^^ the trait `ZeroCopyNew<'_>` is not implemented for `usize`
   |
   = help: the following other types implement trait `ZeroCopyNew<'a>`:
             u16
             u32
             u64
             u8
   = note: required for `Vec<usize>` to implement `ZeroCopyNew<'a>`

error[E0277]: the trait bound `isize: ZeroCopyNew<'_>` is not satisfied
  --> tests/ui/fail/55_usize_isize_types.rs:11:20
   |
11 |     pub opt_isize: Option<isize>,
   |                    ^^^^^^^^^^^^^ the trait `ZeroCopyNew<'_>` is not implemented for `isize`
   |
   = help: the following other types implement trait `ZeroCopyNew<'a>`:
             u16
             u32
             u64
             u8
   = note: required for `Option<isize>` to implement `ZeroCopyNew<'a>`
